### PR TITLE
fix profile validation

### LINKED ISSUE #2721: Profile: Enter " " (space) as school name results in 400 page
If you enter a a space as your school name and hit save the application errors (noticed this on the signup process). Should probably check work experience for a similar bug too.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

fixes #2721

#### What's this PR do?

This ensures that we validate strings composed of just whitespace the same way we do empty strings and null values.

#### How should this be manually tested?

Go to `/profile` and put a whitespace string in any field. You should get a validation error, and it should not make a request to save the profile. You should have to enter a legit value to get it to save.

### CODE REVIEW

**Comment by rhysyngsun** [comment]
@aliceriot could you bump Travis?

File: static/js/lib/validation/profile.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ import { shouldRenderRomanizedFields } from '../../util/profile_edit';
 
 type ErrorMessages = {[key: string]: string};
 
-let isNilOrEmptyString = (val: any): boolean => (
-  val === null || val === undefined || val === ""
+export const isNilOrEmptyString = R.anyPass(
+  [R.isNil, R.equals(""), R.test(/^\s+$/)]
```
Comment (rhysyngsun) [inline static/js/lib/validation/profile.js]: Could simplify to `[R.isNil, R.test(/^\s*$/)]`.

**Comment by rhysyngsun** [comment]
Looks good in manual testing.

**Comment by aliceriot** [comment]
@rhysyngsun I just made that tweak

**Comment by rhysyngsun** [comment]
:+1: assuming tests pass

**Pull request merged**